### PR TITLE
Add lorentz plfc gyrolbn

### PR DESCRIPTION
**Pull request by Longchentong**
Hi Geoopt maintainers,

  I would like to contribute several generic Lorentz-model components derived from our ICLR 2026 paper,
  "Intrinsic Lorentz Neural Network" (ILNN) https://arxiv.org/abs/2602.23981.

  The goal of this PR is not to add the full ILNN model or any experiment-specific training code. Instead,
  I extracted the reusable Lorentz geometry primitives and neural layers that fit Geoopt's scope as
  manifold-aware PyTorch tooling.

  This PR is organized into three reviewable commits:

  1. Add Lorentz gyro primitives

  This commit extends `geoopt.manifolds.lorentz` with Lorentz-native gyro operations and normalization
  helpers:

  - `gyroadd`
  - `gyroinv`
  - `gyroscalar`
  - `lorentz_centroid`
  - `lorentz_dispersion`

  It also adds thin wrapper methods to `geoopt.Lorentz`, following the existing Geoopt style where
  manifold methods delegate to `geoopt.manifolds.lorentz.math`.

  These primitives are useful beyond ILNN, especially for Lorentz-model neural layers and gyrogroup-based
  normalization.

  2. Add Lorentz point-to-hyperplane layer

  This commit adds an experimental Lorentz point-to-hyperplane fully connected layer:

  - `geoopt.layers.LorentzPLFC`
  - alias: `geoopt.layers.PLFC`

  The layer maps Lorentz inputs of shape `(..., in_features + 1)` to Lorentz outputs of shape `(...,
  out_features + 1)`. It uses Geoopt's existing Lorentz curvature convention, where `Lorentz(k)` satisfies
  `<x, x>_L = -k`.

  The implementation is intentionally library-level: it does not include ILNN-specific global state,
  training logic, dataset code, dropout choices, or experiment-specific clamps.

  3. Add GyroLorentz batch normalization

  This commit adds an experimental gyrogroup batch normalization layer:

  - `geoopt.layers.GyroLorentzBatchNorm`
  - alias: `geoopt.layers.GyroLBN`

  The layer uses Lorentz centroid and dispersion statistics, supports running statistics, optional affine
  parameters, and keeps outputs on the Lorentz manifold.

  The implementation is designed as a reusable Lorentz normalization layer rather than a paper-specific
  module.

  Testing:

  I added tests for the new Lorentz primitives and layers, including:

  - Lorentz manifold constraint checks
  - gyro identity/inverse/scalar special cases
  - centroid and dispersion finite-gradient checks
  - PLFC shape, batch-dimension, signed-distance, and backward checks
  - GyroLBN train/eval behavior, running stats, affine=False, small-variance stability, and backward
  checks

  Local verification:

  ```bash
  python -m pytest -q tests/test_lorentz_math.py tests/test_layers_lorentz.py
  # 329 passed

  ruff check geoopt tests
  # All checks passed
  ```

  I also ran the full test suite with PyTorch's legacy torch.load behavior enabled:
  ```bash
  TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD=1 python -m pytest -q tests
  # 9118 passed, 326 skipped
  ```

  Without that environment variable, two existing pickle tests fail due to PyTorch's newer default
  torch.load(weights_only=True), which appears unrelated to this PR.

  I would be happy to adjust naming, API shape, or scope if you prefer keeping only the lower-level
  Lorentz primitives in Geoopt.

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
Hey @Longchentong thanks for the contribution and interest ❤️ ! I'm going to start a copilot review in case I missed anything serious.

**Comment by Longchentong** [comment]
Thanks for the review. I addressed the Copilot comments in e551e0e:
  - added PLFC input shape validation
  - simplified beta using the stable cosh^2 - sinh^2 identity
  - removed the unused time computation in Lorentz gyroadd
  - removed the unused k argument from gyroinv
  - replaced .data usage with .detach()
  - added a wrong-shape PLFC test

**Comment by SauravMaheshkar** [comment]
Hey @Longchentong thanks for addressing the comments. I'm going to proceed to merge this PR.

Would love to get your thoughts on a more generic `geoopt.layers` subpackage as well.
